subsys/net/l2/ethernet/dsa/dsa_port.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(net_dsa_port, CONFIG_NET_DSA_LOG_LEVEL);
+
+#include <zephyr/net/ethernet.h>
+#include <zephyr/net/phy.h>
+#include <zephyr/net/dsa_core.h>
+
+int dsa_port_initialize(const struct device *dev)
+{
+	const struct dsa_port_config *cfg = dev->config;
+	struct dsa_switch_context *dsa_switch_ctx = dev->data;
+	struct net_if *iface = net_if_lookup_by_dev(dev);
+	struct ethernet_context *eth_ctx = net_if_l2_data(iface);
+	struct ethernet_context *eth_ctx_conduit = NULL;
+	int err = 0;
+
+	dsa_switch_ctx->init_ports++;
+
+	/* Find conduit port */
+	if (!dsa_switch_ctx->iface_conduit && cfg->ethernet_connection) {
+		dsa_switch_ctx->iface_conduit = net_if_lookup_by_dev(cfg->ethernet_connection);
+		if (!dsa_switch_ctx->iface_conduit) {
+			LOG_ERR("DSA: Conduit iface NOT found!");
+		}
+
+		eth_ctx->dsa_port = DSA_CPU_PORT;
+
+		/*
+		 * Provide pointer to DSA switch context to conduit's eth interface
+		 * struct ethernet_context
+		 */
+		eth_ctx_conduit = net_if_l2_data(dsa_switch_ctx->iface_conduit);
+		eth_ctx_conduit->dsa_switch_ctx = dsa_switch_ctx;
+		eth_ctx_conduit->dsa_port = DSA_CONDUIT_PORT;
+	}
+
+	if (!cfg->ethernet_connection) {
+		eth_ctx->dsa_port = DSA_USER_PORT;
+		eth_ctx->dsa_switch_ctx = dsa_switch_ctx;
+		dsa_switch_ctx->iface_user[cfg->port_idx] = iface;
+	}
+
+	if (dsa_switch_ctx->dapi->port_init) {
+		err = dsa_switch_ctx->dapi->port_init(dev);
+		if (err) {
+			goto out;
+		}
+	}
+
+out:
+	/* All ports are initialized. May need switch setup. */
+	if (dsa_switch_ctx->init_ports == dsa_switch_ctx->num_ports) {
+		if (dsa_switch_ctx->dapi->switch_setup) {
+			err = dsa_switch_ctx->dapi->switch_setup(dsa_switch_ctx);
+		}
+	}
+
+	return err;
+}
+
+static void dsa_port_iface_init(struct net_if *iface)
+{
+	const struct device *dev = net_if_get_device(iface);
+	struct dsa_port_config *cfg = (struct dsa_port_config *)dev->config;
+	struct dsa_switch_context *dsa_switch_ctx = dev->data;
+	char name[10];
```

Comment:
There is a config option for the length, so we can use it here
```
char name[CONFIG_NET_INTERFACE_NAME_LEN];
```

Reply:
Will do that, thanks.